specification/network/resource-manager/Microsoft.Network/preview/2022-06-01-preview/networkManagerGroupMembership.json

```diff
@@ -0,0 +1,252 @@
+        },
+        {
+          "$ref": "#/parameters/VirtualNetworkNameParameter"
+        },
+        {
+          "$ref": "./network.json#/parameters/ApiVersionParameter"
+        },
+        {
+          "$ref": "#/parameters/FirstIndexParameter"
+        },
+        {
+          "$ref": "#/parameters/PageSizeParameter"
+        }
+      ],
+      "get": {
+        "tags": [
+          "NetworkGroupMemberships_List"
+        ],
+        "operationId": "ListNetworkGroupMemberships",
+        "description": "List networkGroupMemberships applied on a virtual network.",
+        "responses": {
+          "200": {
+            "description": "OK - Returns information about the configurations.",
+            "schema": {
+              "$ref": "#/definitions/NetworkGroupMembershipsListResult"
+            }
+          },
+          "default": {
+            "description": "Resource Provider error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List network group memberships": {
+            "$ref": "./examples/NetworkGroupMembershipsList.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "NetworkGroupMembershipsListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NetworkGroupMembership"
+          },
+          "description": "Gets a page of NetworkGroupMembership"
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Gets the URL to get the next page of results."
+        }
+      },
+      "description": "Result of the request to list NetworkGroupMembership."
+    },
+    "NetworkGroupMembership": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/NetworkGroupMembershipProperties",
+          "description": "The network group membership properties"
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource Location."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v4/types.json#/definitions/Resource"
+        }
+      ],
+      "description": "Resource of networkGroupMembership."
+    },
+    "NetworkGroupMembershipProperties": {
+      "type": "object",
+      "properties": {
+        "groupMemberships": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NetworkGroupMembershipGroup"
+          },
+          "description": "GroupMembership list"
+        }
+      },
+      "description": "Network Group Membership Properties."
+    },
+    "NetworkGroupMembershipGroup": {
```

Comment:
'NetworkGroupMembershipGroup', any need for the trailing 'Group'?

Reply:
It is not named by me, and we have used it for long time, so rename may cause breaking change. Let us still keep it.